frontend/packages/dev-console/src/components/topology/TopologyApplicationResources.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import * as _ from 'lodash';
+import { Link } from 'react-router-dom';
+import { SidebarSectionHeading } from '@console/internal/components/utils';
+import { getActiveNamespace } from '@console/internal/actions/ui';
+import TopologyApplicationResourceList from './TopologyApplicationList';
+import { TopologyDataObject } from './topology-types';
+import { getResourceDeploymentObject } from './topology-utils';
+
+const MAX_RESOURCES = 5;
+
+const ApplicationResource: React.FC<any> = ({ title, kind, resourcesData, group }) => {
+  return (
+    <React.Fragment>
+      {!_.isEmpty(resourcesData) ? (
+        <div className="overview__sidebar-pane-body">
+          <SidebarSectionHeading text={title}>
+            {_.size(resourcesData) > MAX_RESOURCES && (
+              <Link
+                className="sidebar__section-view-all"
+                to={`/search/ns/${getActiveNamespace()}?kind=${kind}&q=${encodeURIComponent(
+                  `app.kubernetes.io/part-of=${group}`,
+                )}`}
+              >
+                {`View All (${_.size(resourcesData)})`}
+              </Link>
+            )}
+          </SidebarSectionHeading>
+          <TopologyApplicationResourceList resources={_.take(resourcesData, MAX_RESOURCES)} />
+        </div>
+      ) : null}
+    </React.Fragment>
+  );
+};
+
+export type TopologyApplicationResourcesProps = {
+  resources: TopologyDataObject[];
+  group: string;
+};
+
+const TopologyApplicationResources: React.FC<TopologyApplicationResourcesProps> = ({
+  resources,
+  group,
+}) => {
+  const resourcesData = {};
+  _.forEach(resources, (res) => {
+    const a = getResourceDeploymentObject(res);
+    resourcesData[a.kind] = [...(resourcesData[a.kind] ? resourcesData[a.kind] : []), a];
+  });
+
+  const renderTitle = (key) => {
+    switch (key) {
+      case 'DeploymentConfig':
+        return 'Deployment Config';
+      case 'Deployment':
+        return 'Deployment';
+      case 'DaemonSet':
+        return 'DaemonSet';
+      case 'StatefulSet':
+        return 'Stateful Set';
+      default:
+        return key;
+    }
+  };
```

Comment:
we could replace this function with model label
```suggestion
  const renderTitle = (kind) => modelFor(kind).label;
```